arbos/block_processor.go

```diff
@@ -214,6 +220,10 @@ func ProduceBlockAdvanced(
 		} else {
 			tx = txes[0]
 			txes = txes[1:]
+			if txesOptions != nil {
+				options = txesOptions[0]
+				txesOptions = txesOptions[1:]
+			}
 			if tx.Type() != types.ArbitrumInternalTxType {
 				hooks = sequencingHooks // the sequencer has the ability to drop this tx
```

Comment:
We should put the options assignment into this if statement, because the conditions don't exist for internal txs. I'd also recommend moving the options into a field of hooks for simplicity.